### ISSUE TITLE
New tile type for selecting a random tile from an atlas without auto tiling.

### ISSUE THREAD

**Issue by CowThing** [godotengine/godot#24549]
**Godot version:**
3.1 Alpha 4

**OS/device including version:**
Windows 64bit

**Issue description:**
I was previously using atlas tiles to create randomly selected tiles that don't need any auto-tiling. For example with this script:

```GDScript
tool
extends TileSet

func _forward_subtile_selection(autotile_id, bitmask, tilemap, tile_location):
	# Atlas Tile ID
	if autotile_id == 0:
		var arr = [Vector2(0, 0), Vector2(0, 1), Vector2(1, 0), Vector2(1, 1)]
		return arr[randi() % arr.size()]
```

But in 3.1 alpha 4 atlas tiles no longer call the function `_forward_subtile_selection`. 

This was more of a workaround on my part since there's no tile type that is just used for selecting a random tile from an atlas. I suggest a new type of random atlas tile that can be used to randomly select a tile from an atlas, as well as call the `_forward_subtile_selection` function (or another similar function). It could also use the priority property to make some tiles more common like auto tiles use.

Here is a simple project of how I am currently doing random tile selection, it uses an auto tile, but ignores the bitmask and just chooses a random tile from the array.

[random_tile_selection.zip](https://github.com/godotengine/godot/files/2705492/random_tile_selection.zip)

**Comment by akien-mga** [comment]
CC @guilhermefelipecgs @YeldhamDev

**Comment by guilhermefelipecgs** [comment]
Well, I needed to disable bitmask for atlas tile because don't make sense for this type of tile mode.

@CowThing  Can you provide an example to me in order to facilitate debug and find a fix for this?

**Comment by guilhermefelipecgs** [comment]
I think you can probably replace atlas_tile to auto_tile. This should works.

**Comment by guilhermefelipecgs** [comment]
> and that would still call `_forward_subtile_selection` like the atlas tile used to.

Maybe it's better to create another method for this, because `_forward_subtile_selection` receive bitmask and this don't fit very well for atlas.

**Comment by CowThing** [comment]
Alright I updated the title and first post, this isn't a bug anymore, but a suggestion for a new random atlas tile that can be used to select a random tile from the atlas. I also uploaded a simple project in the first post showing what I'm doing right now.

If `_forward_subtile_selection` isn't used for the random atlas tile that's fine, but whatever function is used should still have `tile_id`, `tilemap`, and `tile_location` arguments since those are useful.
